From: Sergio Costas Rodriguez <sergio.costas@canonical.com>
Date: Wed, 22 Nov 2023 13:15:42 +0100
Subject: fix possible leak

If pw_check_flatpak() sets app_id, its value will leak when
calling pw_snap_get_audio_permissions(). This patch fixes
this.

(cherry picked from commit ae11e61105d53e4dc0860894ccc17b1a8aead7fa)
Origin: upstream, after 1.0.1
Forwarded: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/1779
---
 src/modules/module-protocol-pulse/server.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/module-protocol-pulse/server.c b/src/modules/module-protocol-pulse/server.c
index 6ae97b7..bbb790f 100644
--- a/src/modules/module-protocol-pulse/server.c
+++ b/src/modules/module-protocol-pulse/server.c
@@ -408,7 +408,7 @@ on_connect(void *data, int fd, uint32_t mask)
 		client_access = server->client_access;
 
 	if (server->addr.ss_family == AF_UNIX) {
-		spa_autofree char *app_id = NULL, *devices = NULL;
+		spa_autofree char *app_id = NULL, *snap_app_id = NULL, *devices = NULL;
 #ifdef HAVE_SNAP
 		pw_sandbox_access_t snap_access;
 #endif
@@ -452,9 +452,9 @@ on_connect(void *data, int fd, uint32_t mask)
 		}
 		// check SNAP permissions
 #ifdef HAVE_SNAP
-		snap_access = pw_snap_get_audio_permissions(client, client_fd, &app_id);
+		snap_access = pw_snap_get_audio_permissions(client, client_fd, &snap_app_id);
 		if ((snap_access & PW_SANDBOX_ACCESS_NOT_A_SANDBOX) == 0) {
-			pw_properties_set(client->props, PW_KEY_SNAP_ID, app_id);
+			pw_properties_set(client->props, PW_KEY_SNAP_ID, snap_app_id);
 
 			pw_properties_set(client->props,
 			                  PW_KEY_SNAP_PLAYBACK_ALLOWED,
